File: frontend/app/src/types/AnnualSimulation.ts
// railsの方の型定義と自動で合わせたいなぁ
export type AnnualSimulation = {
  monthly_deposit: number;
  rate: number;
  year: number;
  simulation_id?: string; // フロントで保存したタイミングではsimulationのid持たないのでoptional
  id?: string; // フロントで保存したタイミングでは自身のid持たないのでoptional
};

File: backend/app/models/annual_simulation.rb
class AnnualSimulation < ApplicationRecord
  belongs_to :simulation
  validates :monthly_deposit, presence: true, length: { maximum: 12 }, numericality: { only_integer: true }
  validates :rate, presence: true, numericality: { only_integer: true }
  validates :years, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than_or_equal_to: 150 }
  validates :order, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than_or_equal_to: 150 }, uniqueness: { scope: :simulation_id }
end
